Make node 8 a package.json requirement

I recently encountered a problem running under Node v6. So... somewhere along the way v8 became a hard requirement. I think there's something like an "engine" requirement we can put in package.json?

+1 to moving straight to Node 8.x. Here's a bit of play-by-play comparing Node 6 👉 7 👉 8:

### Node 6:

```sh
$ npm run lint:js

> firefox-screenshots@22.0.0 lint:js /Users/pdehaan/dev/github/mozilla-services/pageshot
> eslint .

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/webextension/background/analytics.js
  45:36  error  'Object.entries' is not supported yet on Node 6.0.0  node/no-unsupported-features

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/b64.js
  3:16  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  8:16  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/proxy-url.js
   7:40  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  16:42  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/server.js
  867:42  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/servershot.js
  373:17  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  706:19  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/test/addon/browser_screenshots_ui_check.js
  11:10  error  Async functions are not supported yet on Node 6.0.0  node/no-unsupported-features
  12:3   error  Async functions are not supported yet on Node 6.0.0  node/no-unsupported-features
  14:27  error  Async functions are not supported yet on Node 6.0.0  node/no-unsupported-features
  15:5   error  Async functions are not supported yet on Node 6.0.0  node/no-unsupported-features
  19:3   error  Async functions are not supported yet on Node 6.0.0  node/no-unsupported-features

✖ 13 problems (13 errors, 0 warnings)
```

### Node 7

If I bump the engine to Node 7, I still get 12 problems:

```sh
$ npm run lint:js

> firefox-screenshots@22.0.0 lint:js /Users/pdehaan/dev/github/mozilla-services/pageshot
> eslint .

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/b64.js
  3:16  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  8:16  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/proxy-url.js
   7:40  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  16:42  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/server.js
  867:42  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/servershot.js
  373:17  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  706:19  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/test/addon/browser_screenshots_ui_check.js
  11:10  error  Async functions are not supported yet on Node 7.0.0  node/no-unsupported-features
  12:3   error  Async functions are not supported yet on Node 7.0.0  node/no-unsupported-features
  14:27  error  Async functions are not supported yet on Node 7.0.0  node/no-unsupported-features
  15:5   error  Async functions are not supported yet on Node 7.0.0  node/no-unsupported-features
  19:3   error  Async functions are not supported yet on Node 7.0.0  node/no-unsupported-features

✖ 12 problems (12 errors, 0 warnings)
```

### Node 8:

Going straight to Node 8 still gives the `node/no-deprecated-api` errors, but none of the `node/no-unsupported-features` errors:

```sh
$ npm run lint:js

> firefox-screenshots@22.0.0 lint:js /Users/pdehaan/dev/github/mozilla-services/pageshot
> eslint .

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/b64.js
  3:16  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  8:16  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/proxy-url.js
   7:40  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  16:42  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/server.js
  867:42  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/servershot.js
  373:17  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  706:19  error  'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

✖ 7 problems (7 errors, 0 warnings)
```

### Configs ahoy!

```diff
diff --git a/package.json b/package.json
index a0fe0c67..92892a6e 100644
--- a/package.json
+++ b/package.json
@@ -59,6 +59,7 @@
     "eslint": "4.8.0",
     "eslint-plugin-mozilla": "0.4.4",
     "eslint-plugin-no-unsanitized": "2.0.1",
+    "eslint-plugin-node": "5.2.0",
     "eslint-plugin-promise": "3.5.0",
     "eslint-plugin-react": "^7.3.0",
     "fx-runner": "1.0.8",
@@ -76,6 +77,9 @@
     "uglifyify": "4.0.4",
     "web-ext": "2.1.0"
   },
+  "engines": {
+    "node": ">=6.0.0"
+  },
   "homepage": "https://screenshots.firefox.com",
   "license": "MPL-2.0",
   "repository": {
```

```diff
diff --git a/.eslintrc.yml b/.eslintrc.yml
index e5341ba5..82258857 100644
--- a/.eslintrc.yml
+++ b/.eslintrc.yml
@@ -6,6 +6,7 @@ env:
 extends:
   - eslint:recommended
   - plugin:mozilla/recommended
+  - plugin:node/recommended
   - plugin:react/recommended

 parserOptions:
@@ -15,12 +16,27 @@ parserOptions:

 plugins:
   - mozilla
+  - node
   - react
   - promise

 root: true

 rules:
+  node/no-extraneous-require: off
+  node/no-missing-require: off
+  node/no-unpublished-require: off
+  node/shebang: off
+
   react/jsx-uses-react: warn
   react/jsx-uses-vars: warn
   react/no-deprecated: warn # TODO: change to "error"
@@ -31,15 +47,7 @@ rules:
   consistent-return: off # TODO: remove, and default to mozilla/recommended of "error"
   eqeqeq: off # TODO: change to "error"
   no-console: [error, {allow: [debug, error, info, trace, warn]}]
+  no-process-exit: off
   no-var: off # TODO: change to "error"
   prefer-const: off # TODO: change to "error"
   quotes: [off, double] # TODO: change to "error"
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM node:7
+FROM node:8
 
 COPY package.json /app/
 COPY build/server /app/build/server

--- a/package.json
+++ b/package.json
@@ -77,6 +77,9 @@
     "uglifyify": "4.0.4",
     "web-ext": "2.1.0"
   },
+  "engines": {
+    "node": ">=8.0.0"
+  },
   "homepage": "https://screenshots.firefox.com",
   "license": "MPL-2.0",
   "repository": {

